connect/eaas/config.py

```diff
@@ -89,8 +92,29 @@ def get_ws_url(self):
     def get_api_url(self):
         return f'https://{self.env["api_address"]}/public/v1'
 
+    def get_version(self):
+        try:
+            return get_distribution('connect-extension-runner').version
+        except DistributionNotFound:
+            return '0.0.0'
+
+    def get_user_agent(self):
+        version = self.get_version()
+        pimpl = platform.python_implementation()
+        pver = platform.python_version()
+        sysname = platform.system()
+        sysver = platform.release()
+        ua = (
+            f'connect-extension-runner/{version} {self.env["environment_id"]}/'
+            f'{self.env["instance_id"]} {pimpl}/{pver} {sysname}/{sysver}'
+        )
+        return {'User-Agent': ua}
+
     def get_headers(self):
-        return (('Authorization', self.api_key),)
+        return (
+            ('Authorization', self.api_key),
+            ('User-Agent', self.get_user_agent()),
```

Comment:
Customized User-Agent is for ConnectClient instances (both sync and async) not for the websocket.